File: from-0401-to-0500/0402-remove-k-digits/solution.py
class Solution(object):
    def removeKdigits(self, num, k):
        """
        :type num: str
        :type k: int
        :rtype: str
        """
        if k == 0:
            return num

        if k == len(num):
            return "0"

        nums = [int(c) for c in num]

        j = 0
        while j < k:
            idx = 1
            is_deleted = False
            while True:
                prev = nums[idx-1]
    
                if nums[idx] < prev:
                    del nums[idx - 1]
                    is_deleted = True

                idx += 1
                if is_deleted or idx == len(nums):
                    break

            if not is_deleted:
                del nums[-1]

            j += 1

        return str(int(''.join([str(n) for n in nums])))

File: from-0401-to-0500/0402-remove-k-digits/solution.go
func removeKdigits(num string, k int) string {
    digits := strings.Split(num, "")
    stack := []string{}
    for i := 0; i < len(digits); i ++ {
        for k > 0 && len(stack) > 0 && stack[len(stack) - 1] > digits[i] {
            stack = stack[:len(stack) - 1]
            k -= 1
        }
        stack = append(stack, digits[i])
    }
    
    stack = stack[:len(stack) - k]

    lz := 0
    for lz < len(stack) && stack[lz] == "0" {
        lz += 1
    }
    stack = stack[lz:]
    if len(stack) > 0 {
        return strings.Join(stack, "")
    }
    return "0"
}